internal/receiver/simpleprometheusremotewritereceiver/receiver.go

```diff
@@ -0,0 +1,142 @@
+type simplePrometheusWriteReceiver struct {
+	server       *http.Server
+	reporter     internal.Reporter
+	nextConsumer consumer.Metrics
+	cancel       context.CancelFunc
+	config       *Config
+	settings     receiver.CreateSettings
+	sync.Mutex
+}
+
+// newPrometheusRemoteWriteReceiver creates the PrometheusRemoteWrite receiver with the given parameters.
+func newPrometheusRemoteWriteReceiver(
+	settings receiver.CreateSettings,
+	config *Config,
+	nextConsumer consumer.Metrics,
+) (*simplePrometheusWriteReceiver, error) {
+	if nextConsumer == nil {
+		return nil, component.ErrNilNextConsumer
+	}
+
+	rep, err := newReporter(settings)
+	if err != nil {
+		return nil, err
+	}
+
+	r := &simplePrometheusWriteReceiver{
+		settings:     settings,
+		config:       config,
+		nextConsumer: nextConsumer,
+		reporter:     rep,
+	}
+	return r, nil
+}
+func (receiver *simplePrometheusWriteReceiver) buildTransportServer(ctx context.Context, config *internal.ServerConfig) (*http.Server, error) {
+	server, err := internal.NewPrometheusRemoteWriteServer(ctx, config)
+	return server.Server, err
+
+}
+
+// Start starts an HTTP server that can process Prometheus Remote Write Requests
+func (receiver *simplePrometheusWriteReceiver) Start(ctx context.Context, host component.Host) error {
+	metricsChannel := make(chan pmetric.Metrics, receiver.config.BufferSize)
+	cfg := &internal.ServerConfig{
+		HTTPServerSettings: receiver.config.HTTPServerSettings,
+		Path:               receiver.config.ListenPath,
+		Mc:                 metricsChannel,
+		Reporter:           receiver.reporter,
+		Host:               host,
+	}
+	ctx, receiver.cancel = context.WithCancel(ctx)
+	server, err := receiver.buildTransportServer(ctx, cfg)
+	if err != nil {
+		return err
+	}
+	receiver.Lock()
+	defer receiver.Unlock()
+	if nil != receiver.server {
+		err := receiver.server.Close()
+		if err != nil {
+			return err
+		}
+	}
+	receiver.server = server
+	// Start server
+	go func() {
+		if err := receiver.server.ListenAndServe(); err != nil {
+			if !errors.Is(err, net.ErrClosed) {
+				host.ReportFatalError(err)
+			}
+		}
+	}()
+	// Manage server lifecycle
+	go func(ctx2 context.Context) {
+		for {
+			select {
+			case metrics := <-metricsChannel:
+				err := receiver.Flush(ctx2, metrics)
+				if err != nil {
+					receiver.reporter.OnTranslationError(ctx2, err)
+					close(metricsChannel)
+					return
+				}
+			case <-ctx2.Done():
+				close(metricsChannel)
+				return
+			}
+		}
+	}(ctx)
+
+	return nil
+}
+
+// Shutdown stops the PrometheusSimpleRemoteWrite receiver.
+func (receiver *simplePrometheusWriteReceiver) Shutdown(context.Context) error {
+	receiver.Lock()
+	defer receiver.Unlock()
+	if receiver.cancel == nil {
+		return nil
+	}
+	defer receiver.cancel()
+	return receiver.server.Close()
+}
+
+func (receiver *simplePrometheusWriteReceiver) Flush(ctx context.Context, metrics pmetric.Metrics) error {
```

Comment:
Totally understand, but you have to be careful because there are mistakes in the other components 😅. I think a good rule of thumb is that if it doesn't need to be exported, it shouldn't be.